### PR TITLE
Add runtime particle components to HDF5 wrapper

### PR DESCRIPTION
**Pull request by jordanm304**
## Summary

Add runtime components when computing the sizes/offsets for particle HDF5 files. 

## Checklist

The proposed changes:
- [ x] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by atmyers** [comment]
@houjun Does this change look correct to you? Is there a similar change that needs to be made in RestartHDF5 to read the runtime components back in?

**Comment by houjun** [comment]
@atmyers, I just checked the code and this fix looks correct.
The restart code uses "NStructReal + NumRealComps()", and NumRealComps() = NArrayReal + NumRuntimeRealComps(). So no need to change RestartHDF5.
I don't remember why "NStructInt" and "NStructReal" were used, likely a copy-paste error from an earlier version of AMReX.

@jordanm304, to make things consistent, I'd recommend changing the fix to:
```
int_size  = count[grid] * (2 + NStructInt + NumIntComps());
real_size = count[grid] * (AMREX_SPACEDIM + NStructReal + NumRealComps());
```

File: Src/Extern/HDF5/AMReX_ParticleHDF5.H [hunk 1]
```diff
@@ -711,12 +711,13 @@ ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator, CellAssig
         if (count[grid] == 0)
             continue;
 
-        int_size  = count[grid] * (2 + NStructInt + NArrayInt);
+        int_size  = count[grid] * (2 + NStructInt + NArrayInt + NumRuntimeIntComps());
```
Comment (atmyers) [inline Src/Extern/HDF5/AMReX_ParticleHDF5.H]:
```suggestion
        int_size  = count[grid] * (2 + NStructInt + NumIntComps());
```

File: Src/Extern/HDF5/AMReX_ParticleHDF5.H [hunk 2]
```diff
@@ -711,12 +711,13 @@ ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator, CellAssig
...
         my_mfi_int_size.push_back(int_size);
         my_nparticles.push_back(count[grid]);
         my_mfi_int_total_size += int_size;
 
-        real_size = count[grid] * (AMREX_SPACEDIM + NStructReal + NArrayReal);
+
+        real_size = count[grid] * (AMREX_SPACEDIM + NStructReal + NArrayReal + NumRuntimeRealComps() );
```
Comment (atmyers) [inline Src/Extern/HDF5/AMReX_ParticleHDF5.H]:
```suggestion
        real_size = count[grid] * (AMREX_SPACEDIM + NStructReal + NumRealComps());
```

**Review by atmyers — APPROVED** [review]

**Pull request merged**